ambient-mesh/migration/v1/gather-cluster-info.sh

```diff
@@ -0,0 +1,616 @@
+#!/bin/sh
+# Example: CONTEXT="mycluster" ./gather-cluster-info.sh [--hide-names|-hn] [--continue|-c] [--help|-h]
+
+#######################################################################
+# This script collects information about the resources in a Kubernetes context.
+# It is v1 of the script used to get minimal information about the cluster used for v1(?)
+# of the backend which is a generalized overview, without specific details (getting region
+# information, specific _used_ instance cost, etc.)
+#######################################################################
+
+# TODO
+# - get the pods and nodes consecutively instead of processing one, then the other to avoid a mismatch in the metrics data (ex. we get pods info to get metrics, then 2 minutes later get nodes info, but they won't be in a similar state)
+# - add in native sidecars https://istio.io/latest/blog/2023/native-sidecars/
+# - add region information (v2?)
+# - add specific details about the used instances (v2?)
+# - add multi-cluster support (v2?)
+
+#######################################################################
+# Feedback (03/21/2025)
+# - can conditionally use istioctl if it's installed and get resource usage (look into solely relying on kubectl top if resource usage.)
+# - branch to use `istioctl bug report` if it's installed?
+# - TODO: maybe use less files when processing namespaces? it could lead to a lot of tmp file, mayb do a batch process like every 10(0) namespaces, write to final file, then delete tmp files and continue?
+
+# TODO(grand scheme): convert this to go binary / keep as script in the meantime
```

Comment:
Can we break these out into GH issues so that it'll be easier to track?

Reply:
Created a few issues 🫡 https://github.com/solo-io/scripts-public/issues